### PR TITLE
Fixing transformations to accept theano tensors - GPU testing

### PR DESCRIPTION
**Pull request by springcoil**
removed unnecessary file

- This is a test branch I did to try to get this to work.

### CODE REVIEW

File: pymc3/examples/baseball.py [hunk 1]
```diff
@@ -5,9 +5,11 @@
 
 import pymc3 as pm
 import numpy as np
+import theano
 
 data = np.loadtxt( 'data/efron-morris-75-data.tsv', delimiter="\t", skiprows=1, usecols=(2,3) )
 
+theano.config.floatX = 'float32'
```
Comment (twiecki) [inline pymc3/examples/baseball.py]: this is set in your config file, we shouldn't set it here.

File: pymc3/examples/baseball.py [hunk 2]
```diff
@@ -5,9 +5,11 @@
...
 atBats = data[:,0]
```
Comment (twiecki) [inline pymc3/examples/baseball.py]: I would just do the .astype() here.

File: pymc3/examples/baseball.py [hunk 3]
```diff
@@ -20,9 +22,9 @@
 
 with model:
     phi = pm.Uniform( 'phi', lower=0.0, upper=1.0 )
-    kappa = BoundedKappa( 'kappa', alpha=1.0001, m=1.5 )
-    thetas = pm.Beta( 'thetas', alpha=phi*kappa, beta=(1.0-phi)*kappa, shape=N )
-    ys = pm.Binomial( 'ys', n=atBats, p=thetas, observed=hits )
+    kappa = BoundedKappa( 'kappa', alpha=1.0001,  m=1.5 )
+    thetas = pm.Beta( 'thetas', alpha=phi.astype(theano.config.floatX)*kappa.astype(theano.config.floatX) , beta=((1.0-phi)*kappa).astype(theano.config.floatX), shape=N )
```
Comment (twiecki) [inline pymc3/examples/baseball.py]: You shouldn't need `.astype()` for RVs, only numpy arrays.

**Comment by springcoil** [comment]
Ok I fixed that all and I still get the same errors.

Am I missing something I'm using a local branch I've run python setup.py develop and I've made those changes like you said and I still get errors...

**Comment by twiecki** [comment]
Could be somewhere else where we are introducing this. I would set http://deeplearning.net/software/theano/library/config.html#config.warn_float64 to raise and then use a debugger.

**Comment by twiecki** [comment]
Can you push?

Also, I think I know: https://github.com/pymc-devs/pymc3/blob/9992e48168f3364ce2d8643f9b79acc13b6e79b7/pymc3/examples/baseball.py#L21 is causing this.

We need to add `tt.as_theano_tensor()` to the inputs of the transforms: https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/transforms.py#L141 That way, the values that get passed in (float64 in this case) get transformed to floatX.

**Comment by twiecki** [comment]
You can confirm by changing:
`BoundedKappa = pm.Bound( pm.Pareto, lower=pm.theanof.floatX(1.0))`

**Comment by springcoil** [comment]
I pushed - that's what I thought too. Let me confirm that.

**Comment by springcoil** [comment]
I confirm that it appears to be working @twiecki with that change. 
However it takes a looong time.

**Comment by springcoil** [comment]
I made a few updates to the transform variables. I confirm it works now.

File: pymc3/examples/baseball.py [hunk 4]
```diff
@@ -20,8 +21,8 @@
 
 with model:
     phi = pm.Uniform( 'phi', lower=0.0, upper=1.0 )
-    kappa = BoundedKappa( 'kappa', alpha=1.0001, m=1.5 )
-    thetas = pm.Beta( 'thetas', alpha=phi*kappa, beta=(1.0-phi)*kappa, shape=N )
+    kappa = BoundedKappa( 'kappa', alpha=1.0001,  m=1.5 )
```
Comment (twiecki) [inline pymc3/examples/baseball.py]: unneeded spaces

File: pymc3/examples/baseball.py [hunk 5]
```diff
@@ -20,8 +21,8 @@
...
+    thetas = pm.Beta( 'thetas', alpha=phi*kappa , beta=(1.0-phi)*kappa, shape=N )
```
Comment (twiecki) [inline pymc3/examples/baseball.py]: same

File: pymc3/examples/baseball.py [hunk 6]
```diff
@@ -20,7 +21,7 @@
 
 with model:
     phi = pm.Uniform( 'phi', lower=0.0, upper=1.0 )
-    kappa = BoundedKappa( 'kappa', alpha=1.0001, m=1.5 )
+    kappa = BoundedKappa( 'kappa', alpha=1.0001,  m=1.5 )
```
Comment (twiecki) [inline pymc3/examples/baseball.py]: `kappa = BoundedKappa('kappa', alpha=1.0001, m=1.5 )`

**Comment by springcoil** [comment]
I think that should be ready to go.

**Comment by springcoil** [comment]
This should resolve https://github.com/pymc-devs/pymc3/issues/2032

**Pull request merged**